### ISSUE TITLE
`napi_get_property_names` is buggy in node 10.19 

### ISSUE THREAD

**Issue by artemp** [mapbox/vtshaver#42]
```c++
napi_get_property_names(_env, _value, &result);
```
erroneously returns `Number` type when property name has only digit literals e.g '123' 

This has been fixed in node >=12 (tested 12.16.1, 13.12.0)
/cc @springmeyer

**Comment by springmeyer** [comment]
@artemp nice catch. Can you try to track down what change in node core fixed what you are seeing? Also, could this explain https://github.com/mapbox/vtshaver/pull/40 ?

**Comment by springmeyer** [comment]
> Also, could this explain #40

This btw seemed to me like a v8 behavior rather than node core or Nan related, but I'm not positive.

**Comment by artemp** [comment]
yep, https://github.com/mapbox/vtshaver/issues/42 looks related. I'll investigate further and add workaround in our code if required. 
/cc @springmeyer

**Comment by springmeyer** [comment]
thanks @artemp - I feel pretty good about #40 as a workaround, but let me know if you see a better one.

**Comment by artemp** [comment]
@springmeyer - re: https://github.com/mapbox/vtshaver/pull/40 fix - ideally we shouldn't use `v8` methods from N-api module.

**Comment by springmeyer** [comment]
> ideally we shouldn't use v8 methods from N-api module.

Ah, good point.

**Comment by artemp** [comment]
#### Fix 

```c++
Napi::String Napi::Value::ToString() const;
```
^^ Returns the Napi::Value coerced to a JavaScript string.
